Xplat/src/main/java/vazkii/patchouli/common/util/ItemStackUtil.java

```diff
@@ -55,32 +62,34 @@ public static Triple<ResourceLocation, Integer, CompoundTag> parseItemStackStrin
 
 		ResourceLocation key = new ResourceLocation(tokens[0], tokens[1]);
 		int countn = Integer.parseInt(count);
-		CompoundTag tag = null;
+		DataComponentMap.Builder componentMap = DataComponentMap.builder();
 
-		if (!nbt.isEmpty()) {
+		if (!components.isEmpty()) {
+			ItemParser parser = new ItemParser(VanillaRegistries.createLookup());
 			try {
-				tag = TagParser.parseTag(nbt);
+				ItemParser.ItemResult result = parser.parse(new StringReader(res.toString() + components)); //TODO: Check if the parsed result is correct
+				componentMap.addAll(result.components());
 			} catch (CommandSyntaxException e) {
 				throw new RuntimeException("Failed to parse ItemStack JSON", e);
 			}
 		}
 
-		return ImmutableTriple.of(key, countn, tag);
+		return ImmutableTriple.of(key, countn, componentMap.build());
```

Comment:
1. use the patch directly not the map
2. this method checks for `{` for the components but the components use `[` now

Reply:
I don't know regex 😓

Reply:
It's probably the `res.indexOf("{")` isn't it 😓